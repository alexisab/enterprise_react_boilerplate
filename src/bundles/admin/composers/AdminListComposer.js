import { pure, compose, withHandlers } from 'recompose';
import { graphql } from 'react-apollo';
import { adminListQuery, sendInviteMutation } from '../graphql';
import { displayLoadingState } from '../../../helpers/compose';
import { Notification } from '../../../shared';

const AdminListComposer = compose(
  graphql(adminListQuery),
  displayLoadingState,
  graphql(sendInviteMutation, { name: 'sendInviteMutate'} ),
  withHandlers({
    handleDelete: props => (id) => {
      console.log('handleDelete');
    },
    sendInvite: props => (id) => {
      const { sendInviteMutate } = props;
      sendInviteMutate({
        variables: { id }
      }).then(({ data: { sendInvite } }) => {
        if (sendInvite.errors.length <= 0) {
          Notification('success', 'Admin Invited', `We have send another invitation email to ${sendInvite.name}.`);
        }
      })
      console.log('send Invite')
    }
  }),
  pure
);

export default AdminListComposer;
