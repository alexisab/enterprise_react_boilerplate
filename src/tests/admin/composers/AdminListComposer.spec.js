import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
// import { setMockGraphQLProps } from "react-apollo";
import AdminListComposer from "../../../bundles/admin/composers/AdminListComposer";

const EmptyComponent = () => null;
const EnhancedComponent = AdminListComposer(props => (
  <EmptyComponent {...props} />
));

describe("when admins are fetched", () => {
  test("it should match snapshot", () => {
    const wrapper = mount(<EnhancedComponent />);
    const enhancedComponent = wrapper.find(EmptyComponent);
    expect(enhancedComponent.props()).toMatchSnapshot();
  });

  test("it should rendered the 2 admins", () => {
    const wrapper = mount(<EnhancedComponent />);
    const enhancedComponent = wrapper.find(EmptyComponent);
    const adminList = enhancedComponent.props().data;
    expect(adminList.admins.length).toEqual(2);
  });
});
