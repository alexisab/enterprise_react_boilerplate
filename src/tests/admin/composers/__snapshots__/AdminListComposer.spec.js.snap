// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adminListComposer when admins are fetched successfully it should match snapshot 1`] = `
Object {
  "data": Object {
    "admins": Array [
      Object {
        "admin": true,
        "designation": null,
        "email": "admin1@eg.com",
        "errors": Array [],
        "id": 1,
        "name": "admin1",
        "status": "active",
        "super_admin": false,
      },
      Object {
        "admin": true,
        "designation": null,
        "email": "admin2@eg.com",
        "errors": Array [],
        "id": 2,
        "name": "admin2",
        "status": "active",
        "super_admin": false,
      },
    ],
  },
  "handleDelete": [Function],
  "sendInvite": [Function],
  "sendInviteMutate": [Function],
}
`;

exports[`adminListComposer when there is an error in fetching admins it should match snapshot 1`] = `
Object {
  "data": Object {
    "admins": Array [
      Object {
        "admin": true,
        "designation": null,
        "email": "admin1@eg.com",
        "errors": Array [
          Object {
            "key": "error",
            "message": "error message",
          },
        ],
        "id": 1,
        "name": "admin1",
        "status": "active",
        "super_admin": false,
      },
      Object {
        "admin": true,
        "designation": null,
        "email": "admin2@eg.com",
        "errors": Array [],
        "id": 2,
        "name": "admin2",
        "status": "active",
        "super_admin": false,
      },
    ],
  },
  "handleDelete": [Function],
  "sendInvite": [Function],
  "sendInviteMutate": [Function],
}
`;
